src/Utilities/Gsl.hpp

```diff
@@ -131,6 +131,13 @@ SPECTRE_ALWAYS_INLINE constexpr const T& at(std::initializer_list<T> cont,
 }
 // @}
 
+/*!
+ * \ingroup UtilitiesGroup
+ * \brief Mark a raw pointer as owning its data
+ */
+template <class T, Requires<std::is_pointer<T>::value> = nullptr>
+using owner = T;
```

Comment:
clang-tidy 6.0 requests this, so we certainly should not get rid of it. I think we should handle not wanting owning raw pointers in one of two ways:
- Have a static_assert be triggered if `owner` is used so that when clang-tidy complains, someone uses and hits a static_assert (plus documentation)
- We only add documentation discouraging it.

I'm fine with either, but since we really won't ever have owning raw pointers the first is better.

I noticed we were missing owner because of clang-tidy complaining inside the pup function for unique_ptr.